cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1293,14 +1335,21 @@ class ObjectOutputStream final : public io::OutputStream {
   }
 
   Status Flush() override {
+    if (closed_) {
+      return Status::Invalid("Operation on closed stream");
+    }
```

Comment:
Is this check needed?  If `closed_` is true then `FlushAsync` is just going to immediately return a failed future anyways.